konrad/cloud.py

```diff
@@ -9,145 +9,136 @@
-                0.85*np.ones((1, num_sw_bands, 1, numlevels)),
-                dims=('longitude', 'num_shortwave_bands',
-                      'latitude', 'mid_levels'),
-                attrs={'units': 'dimensionless'})
-        if sw_optical_thickness is not None:
-            self.shortwave_optical_thickness_due_to_cloud = sw_optical_thickness
-        else:
-            self.shortwave_optical_thickness_due_to_cloud = DataArray(
-                np.zeros((1, num_sw_bands, 1, numlevels)),
-                dims=('longitude', 'num_shortwave_bands',
-                      'latitude', 'mid_levels'),
-                attrs={'units': 'dimensionless'})
-        if single_scattering_albedo is not None:
-            self.single_scattering_albedo_due_to_cloud = single_scattering_albedo
-        else:
-            self.single_scattering_albedo_due_to_cloud = DataArray(
-                0.9*np.ones((1, num_sw_bands, 1, numlevels)),
-                dims=('longitude', 'num_shortwave_bands',
-                      'latitude', 'mid_levels'),
-                attrs={'units': 'dimensionless'})
+        self.cloud_ice_particle_size = get_p_data_array(ice_particle_size,
+                                                        numlevels=numlevels,
+                                                        units='micrometers')
+
+        self.cloud_water_droplet_radius = get_p_data_array(droplet_radius,
+                                                           numlevels=numlevels,
+                                                           units='micrometers')
+
+        self.longwave_optical_thickness_due_to_cloud = get_waveband_data_array(
+                lw_optical_thickness, numlevels=numlevels,
+                numbands=num_lw_bands, sw=False)
+
+        self.cloud_forward_scattering_fraction = get_waveband_data_array(
+                forward_scattering_fraction, numlevels=numlevels,
+                numbands=num_sw_bands)
+
+        self.cloud_asymmetry_parameter = get_waveband_data_array(
+                asymmetry_parameter, numlevels=numlevels,
+                numbands=num_sw_bands)
+
+        self.shortwave_optical_thickness_due_to_cloud = \
+                get_waveband_data_array(sw_optical_thickness,
+                                        numlevels=numlevels,
+                                        numbands=num_sw_bands)
+
+        self.single_scattering_albedo_due_to_cloud = get_waveband_data_array(
+                single_scattering_albedo, numlevels=numlevels,
+                numbands=num_sw_bands)
 
     @abc.abstractmethod
     def update_cloud_profile(self, **kwargs):
         """Return the cloud parameters for the radiation scheme."""
 
 
 class ClearSky(Cloud):
-    def __init__(self, numlevels=200):
-        """Clear-sky. Set cloud area fraction to zero everywhere.
+    def __init__(self):
+        """Clear-sky. Take defaults from parent class, with cloud area fraction
+        equal to zero everywhere."""
+        super().__init__()
```

Comment:
If you remove the `__init__` method, it will use the one defined in the parent class. You can move the documentation to the class docstring.